KioskApplication/src/KioskApplication/entity/MapEntity.java

```diff
@@ -0,0 +1,73 @@
+package KioskApplication.entity;
+
+import KioskApplication.database.objects.Edge;
+import KioskApplication.database.objects.Node;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class MapEntity implements IMapEntity {
```

Comment:
This is a good implementation. Can you make this a singleton so a single instance can be used across the program.

Typically singletons have a structure that looks like this which allows for **only one instance** to exist during runtime:
```
private static MapEntity _instance = null;
public static MapEntity getInstance() {
   if (_instance == null) instance = new MapEntity(...);
   return _instance;
}
```